src/__tests__/useMap.test.ts

```diff
@@ -106,3 +106,14 @@ it('should reset map to initial object provided', () => {
 
   expect(result.current[0]).toEqual({ foo: 'bar', a: 1 });
 });
+
+it('should memoized its utils methods', () => {
+  const { result } = setUp({ foo: 'bar', a: 1 });
+  const [, utils] = result.current;
+
+  act(() => {
+    utils.set('foo', 'baz');
+  });
+
+  expect(result.current[1]).toBe(utils);
```

Comment:
You are checking here that `utils` object is the same instance, but that doesn't mean that returned callbacks are same instances, so you would have to check that `utils.get` is `result.current[1].get` and so on.

Reply:
In fact it does mean that internals are the same (due to `useMemo` instead of bunch of `useCallback`), but that, more precise check of `result.current[1].get` wont be excessive.

Reply:
It is indeed working just fine, but I guess an additionnal test assertion can make it more visible